src/wp-includes/widgets/class-wp-widget-media-image.php

```diff
@@ -239,14 +239,31 @@ public function render_media( $instance ) {
 				$instance['height'] = '';
 			}
 
-			$image = sprintf(
-				'<img class="%1$s" src="%2$s" alt="%3$s" width="%4$s" height="%5$s" />',
-				esc_attr( $classes ),
-				esc_url( $instance['url'] ),
-				esc_attr( $instance['alt'] ),
-				esc_attr( $instance['width'] ),
-				esc_attr( $instance['height'] )
+			$attr = array(
+				'class'    => $classes,
+				'src'      => $instance['url'],
+				'alt'      => $instance['alt'],
+				'width'    => $instance['width'],
+				'height'   => $instance['height'],
+				'decoding' => 'async',
 			);
+
+			$loading_optimization_attr = wp_get_loading_optimization_attributes(
+				'img',
+				$attr,
+				'widget_media_image'
+			);
+
+			$attr = array_merge( $attr, $loading_optimization_attr );
+
+			$attr  = array_map( 'esc_attr', $attr );
+			$image = '<img';
+
+			foreach ( $attr as $name => $value ) {
+				$image .= ' ' . $name . '="' . $value . '"';
+			}
```

Comment:
This would add sanitizing for the `src` as well as move escaping closer to the point of output:

```suggestion
			$image       = '<img';
			$attr['src'] = esc_url_raw( $attr['src'] );
			foreach ( $attr as $name => $value ) {
				$image .= ' ' . $name . '="' . esc_attr( $value ) . '"';
			}
```

Reply:
To @mukeshpanchal27 's point, this is what `wp_get_attachment_image` had does.  https://github.com/WordPress/wordpress-develop/pull/4786/files#r1253031188

Reply:
I rescind my suggestion, as `esc_url_raw()` does not like `data:` URLs:

```
wp> esc_url_raw('data:image/gif;base64,R0lGODlhAQABAAAAACw=')
=> string(0) ""
```